samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -65,12 +66,10 @@
   private volatile String ephemeralPath = null;
   private final ZkKeyBuilder keyBuilder;
   private final int connectionTimeoutMs;
-  private ZkJobCoordinatorMetrics metrics;
+  private final ZkJobCoordinatorMetrics metrics;
 
   public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
-    this.keyBuilder = zkKeyBuilder;
-    this.connectionTimeoutMs = connectionTimeoutMs;
-    this.zkClient = zkClient;
+    this (zkKeyBuilder, zkClient, connectionTimeoutMs, new ZkJobCoordinatorMetrics(new MetricsRegistryMap()));
```

Comment:
Is this constructor used only for testing? 

If not, it should not be creating a separate ZkJobCoordinatorMetrics instance with a new metrics registry. You should wire-in the MetricsRegistry from the caller and use it in a `ZkUtilsMetrics` class, although I don't see a value of separating the metrics encapsulating class here. Alternative is to wire-in the metrics instance from the caller itself and remove this constructor.